### PR TITLE
Fix table element parsing in ie11

### LINKED ISSUE #3: <template> polyfill not working when template root element is <tr>
_From @herrernst on October 6, 2016 9:33_

Hi,
I was looking for just a polyfill for the HTML5 `<template>` tag. I've tried to use yours but it doesn't give the expected result when the `<template>` element contains a `<tr>`. There is an example adopted from [MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/template) here: 
https://codepen.io/superlupo/pen/qapgZY
This works in Chrome and Firefox, but not in IE (11). I suspect this is because the HTML parser throws away the `<tr>` and `<td>` when they are not inside of a `<table>`; so the template then is basically empty. I reproduced this also with another polyfill. Can you think of any way to work around that? If this is a general restriction on all `<template>` polyfills, then maybe this should be noted.
Thanks

_Copied from original issue: webcomponents/webcomponentsjs#632_

### PR DESCRIPTION
**Pull request by TimvdLippe**
Using the fix from jQuery when setting the `innerHTML` text. See the links in the source code for the original implementation. This solution was previously submitted in a different form in #16 and #19 by @kapouer, but these were never merged for other unrelated reasons.

Fixes #3

### CODE REVIEW

**Review by sorvell — CHANGES_REQUESTED** [review]
Let's add tests for all the special table nodes (col and thead look missing). Otherwise looks ok. Thanks!

**Comment by TimvdLippe** [comment]
@sorvell Done!

**Comment by jay8t6** [comment]
@TimvdLippe This is awesome! Thanks!

**Comment by web-padawan** [comment]
I'm wondering if the similar fix can be applied to the `select` element as well?
See Polymer/polymer#1735

**Comment by TimvdLippe** [comment]
@web-padawan Hm yes that should be possible. Let me update the PR tomorrow

File: template.js [hunk 1]
```diff
@@ -229,12 +229,12 @@
 
     // Taken from https://github.com/jquery/jquery/blob/73d7e6259c63ac45f42c6593da8c2796c6ce9281/src/manipulation/wrapMap.js
     var topLevelWrappingMap = {
-      ['option']: ['select'],
-      ['thead']: ['table'],
-      ['col']: ['colgroup', 'table'],
-      ['tr']: ['tbody', 'table'],
-      ['th']: ['tr', 'tbody', 'table'],
-      ['td']: ['tr', 'tbody', 'table']
+      'option': ['select'],
```
Comment (web-padawan) [inline template.js]: BTW maybe also `optgroup` tag?

Comment (TimvdLippe) [reply template.js]: I think that is working correctly in IE11, is it not?

Comment (web-padawan) [reply template.js]: Not sure, I was just guessing. Here is the glitch to check: https://morning-fuel.glitch.me

**Comment by smalluban** [comment]
@sorvell Is there anything more, that stops this PR from merging?

**Review by dfreedm — APPROVED** [review]

**Comment by TechQuery** [comment]
But it will be failed in this case:
```HTML
<table>
    <tbody><template>
        <tr><td>Something...</td></tr>
    </template></tbody>
</table>
```
**HTML parser of IE** will drop `<template />` directly, you won't find anything of your template...

**Comment by TimvdLippe** [comment]
@TechQuery Yes and there is nothing we can do about it :cry: The `template` is already gone before we even have a chance to search for it. For more info see: https://github.com/Polymer/polymer/issues/1567#issuecomment-381980212

**Comment by TechQuery** [comment]
@TimvdLippe So I can only allow users to [replace `<template />` with `<!-- -->`](https://github.com/EasyWebApp/WebCell/blob/61a301c64756249103d072e5882b822d1397d50d/source/view/ArrayView.js#L27)...

**Comment by TimvdLippe** [comment]
Unfortunately that is the result of parsing of IE11. Don't think you have any other option. I would like it to be different, but haven't found a workable solution :cry:

**Comment by tekreme73** [comment]
I got a workable solution for you if you have to use a table layout.

I created **a custom element for each layers** of my table: `x-table`, `x-header`, `x-row` and `x-cell`. For each layers, I defined the `:host` CSS `display` property to match the table layout behavior like:
- `x-table` => `display: table`
- `x-header` => `display: table-header-group`
- `x-row` => `display: table-row`
- `x-cell` => `display: table-cell`

With this solution, I'm able to do `dom-repeat` for the rows, even with IE11.
It **will need some CSS tricks** to have the same behavior as a native `table` tag but it does the stuff.

**Pull request merged**